src/plugins/controls/public/range_slider/embeddable/range_slider_embeddable.tsx

```diff
@@ -170,24 +173,47 @@ export class RangeSliderEmbeddable extends Embeddable<RangeSliderEmbeddableInput
     // fetch available min/max when input changes
     this.subscriptions.add(
       dataFetchPipe.subscribe(async () =>
-        this.runRangeSliderQuery().catch((e) => {
-          this.dispatch.setErrorMessage(e.message);
-        })
+        this.runRangeSliderQuery()
+          .catch((e) => {
+            this.dispatch.setErrorMessage(e.message);
+          })
+          .then(async () => {
+            await this.buildFilter();
+          })
       )
     );
 
-    // build filters when value change
+    // write to unpublished changes when value changes
     this.subscriptions.add(
       this.getInput$()
-        .pipe(
-          debounceTime(400),
-          distinctUntilChanged((a, b) => isEqual(a.value, b.value)),
-          skip(1) // skip the first input update because initial filters will be built by initialize.
-        )
-        .subscribe(this.buildFilter)
+        .pipe(distinctUntilChanged((a, b) => isEqual(a.value, b.value)))
+        .subscribe(({ value }) => {
+          this.unpublishedChanges.next({ range: value ?? ['', ''] });
```

Comment:
We need this so that, if the selected value is `undefined` in the new input, we want to treat it as "no selections" - otherwise, the early return in `publishNewRange()` will prevent these input changes from actually following through (for example, resetting the dashboard could cause a **valid** reset from some selected value to `undefined`).